lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+  push @{$self->static->paths},   $home->child('public')->to_string;
+
+  # Default to controller and application namespace
+  my $controller = "@{[ref $self]}::Controller";
+  my $r          = $self->preload_namespaces([$controller])->routes->namespaces([$controller, ref $self]);
+
+  $self->plugin($_) for qw(HeaderCondition DefaultHelpers TagHelpers EPLRenderer EPRenderer);
+
+  # Exception handling should be first in chain
+  $self->hook(around_dispatch => \&_exception);
+
+  $self->startup;
+  $self->warmup;
+
+  return $self;
+}
+
+sub plugin {
+  my $self = shift;
+  $self->plugins->register_plugin(shift, $self, @_);
+}
+
+sub server { $_[0]->plugins->emit_hook(before_server_start => @_[1, 0]) }
+
+sub start {
+  my $self = shift;
+  $_->warmup for $self->static, $self->renderer;
+  return $self->commands->run(@_ ? @_ : @ARGV);
+}
+
+sub startup { }
+
+sub warmup { Mojo::Loader::load_classes $_ for @{shift->preload_namespaces} }
+
+sub _action {
+  my ($next, $c, $action, $last) = @_;
+  my $val = $action->($c);
+  $val->catch(sub { $c->helpers->reply->exception(shift) }) if Scalar::Util::blessed $val && $val->isa('Mojo::Promise');
+  return $val;
+}
+
+sub _die { CORE::die ref $_[0] ? $_[0] : Mojo::Exception->new(shift)->trace }
+
+sub _exception {
+  my ($next, $c) = @_;
+  local $SIG{__DIE__} = \&_die;
+  $c->helpers->reply->exception($@) unless eval { $next->(); 1 };
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Mojolicious - Real-time web framework
+
+=head1 SYNOPSIS
+
+  # Application
+  package MyApp;
+  use Mojo::Base 'Mojolicious', -signatures;
+
+  # Route
+  sub startup ($self) {
+    $self->routes->get('/hello')->to('foo#hello');
+  }
+
+  # Controller
+  package MyApp::Controller::Foo;
+  use Mojo::Base 'Mojolicious::Controller', -signatures;
+
+  # Action
+  sub hello ($self) {
+    $self->render(text => 'Hello World!');
+  }
+
+=head1 DESCRIPTION
+
+An amazing real-time web framework built on top of the powerful L<Mojo> web development toolkit. With support for
+RESTful routes, plugins, commands, Perl-ish templates, content negotiation, session management, form validation,
+testing framework, static file server, C<CGI>/C<PSGI> detection, first class Unicode support and much more for you to
+discover.
+
+Take a look at our excellent documentation in L<Mojolicious::Guides>!
+
+=head1 HOOKS
+
+L<Mojolicious> will emit the following hooks in the listed order.
+
+=head2 before_command
+
+Emitted right before the application runs a command through the command line interface.
+
+  $app->hook(before_command => sub ($command, $args) {...});
+
+Useful for reconfiguring the application before running a command or to modify the behavior of a command. (Passed the
+command object and the command arguments)
+
+=head2 before_server_start
+
+Emitted right before the application server is started, for web servers that support it, which includes all the
+built-in ones.
+
+  $app->hook(before_server_start => sub ($server, $app) {...});
```

Comment:
It's easy to determine if the app was started with daemon or prefork.  Can it be possible to detect if the app was started with Morbo or Hypnotoad?  Perhaps there's no reason for an app to know, but why not provide the information if possible?